scimath_f/lawson.f

```diff
@@ -993,15 +1051,16 @@ SUBROUTINE DNNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
 C
       IF (JJ.EQ.NSETP) GO TO 290
       JJ=JJ+1
-          DO 280 J=JJ,NSETP
+        DO J=JJ,NSETP
           II=INDEX(J)
           INDEX(J-1)=II
           CALL DG1 (A(J-1,II),A(J,II),CC,SS,A(J-1,II))
           A(J,II)=ZERO
               DO 270 L=1,N
```

Comment:
Why did you not replace this with a line number less `DO`? I.e. replace this with
```
DO L=1,N
  IF (L.NE.II) CALL DG2 (CC,SS,A(J-1,L),A(J,L))
END DO
```

Reply:
As discussed offline, not all DO's need to be replaced with the `DO` / `END DO` form.